packages/libraries/federation-link-utils/src/index.ts

```diff
@@ -39,20 +38,27 @@ export function extractLinkImplementations(typeDefs: DocumentNode): {
   const linkByIdentity = Object.fromEntries(
     FederatedLink.fromTypedefs(typeDefs).map(l => [l.identity, l]),
   );
-  const supportsFederationV2 = linkByIdentity[FEDERATION_IDENTITY] !== undefined;
+  // Any schema with a `@link` directive present is considered federation 2
+  // although according to federation docs, schemas require linking specifically
+  // the federation 2.x spec. The reason for not being so picky is that supergraphs also
+  // use @link, but do not necessarily link to the federation 2.x spec.
+  const supportsFederationV2 = Object.keys(linkByIdentity).length > 0;
```

Comment:
I'm curious what others think about this.

Another option is to not support federation 1 at all, or have a separate name resolver function for fed1.